internal/services/compute/linux_virtual_machine_scale_set_resource.go

```diff
@@ -230,7 +225,17 @@ func resourceLinuxVirtualMachineScaleSetCreate(d *pluginsdk.ResourceData, meta i
 	}
 
 	if v, ok := d.Get("extension_operations_enabled").(bool); ok {
-		virtualMachineProfile.OsProfile.AllowExtensionOperations = utils.Bool(v)
+		if v && !provisionVMAgent {
+			return fmt.Errorf("`extension_operations_enabled` cannot be set to `true` when `provision_vm_agent` is set to `false`")
+		}
+
+		if !features.FourPointOhBeta() {
+			if !pluginsdk.IsExplicitlyNullInConfig(d, "extension_operations_enabled") {
+				virtualMachineProfile.OsProfile.AllowExtensionOperations = utils.Bool(v)
+			}
```

Comment:
When not specified, value of `extension_operations_enabled` is set by service, and when `provision_vm_agent` is true, it will be set to true, when `provision_vm_agent` is false, it will be set to false. As this is a newly introduced property, to avoid breaking change, only send its value when it's not nil in the config. In 4.0, it will have default value `true`.